src/main/java/io/fabric8/elasticsearch/plugin/acl/DynamicACLFilter.java

```diff
@@ -238,98 +274,135 @@ private synchronized void syncAcl() {
+                .consistencyLevel(WriteConsistencyLevel.DEFAULT)
+                .source(rolesMapping.toMap());
+		
+		String rmID = esClient.index(mappingIR).actionGet().getId();
+		logger.debug("Built rolesMapping request: '{}'", mappingIR);
+		logger.debug("rolesMapping ID: '{}'", rmID);
+
+		// force a config reload
+		String[] updateString = {"roles", "rolesmapping"};		
+		ConfigUpdateResponse cur = esClient.execute(ConfigUpdateAction.INSTANCE, new ConfigUpdateRequest(updateString)).actionGet();
+		
+		if ( cur.getNodes().length > 0 )
+			logger.debug("Successfully reloaded config with '{}' nodes", cur.getNodes().length);
+		else
+			logger.warn("Failed to reloaded config -- '{}' nodes updated", cur.getNodes().length);
 	}
 	
-	private void seedInitialACL(Client esClient) throws Exception {
+	private void createConfig(Client esClient) throws IOException {
+		
+		final ClusterHealthResponse chr = esClient.admin().cluster().health(new ClusterHealthRequest().waitForYellowStatus()).actionGet();
 
-		SearchGuardACL acl = new SearchGuardACL();
-		boolean create = false;
+        if ( ! chr.isTimedOut() ) {
+		
+			esClient.admin().indices().create( new CreateIndexRequest().index(searchGuardIndex)
+					.settings("index.number_of_shards", 1, "index.number_of_replicas", chr.getNumberOfDataNodes()-1)
+	                ).actionGet().isAcknowledged();
+
+			// Wait for health status of YELLOW
+			ClusterHealthRequest healthRequest = new ClusterHealthRequest()
+					.indices(new String[]{searchGuardIndex}).waitForYellowStatus();
+			
+			esClient.admin().cluster().health(healthRequest).actionGet().getStatus();
+			
+			String cd = settings.get(SG_CONFIG_SETTING_PATH, DEFAULT_SG_CONFIG_SETTING_PATH);
+			// create initial sg config
+			boolean success = uploadFile(esClient, cd+"sg_config.yml", "config");
+            success = success & uploadFile(esClient, cd+"sg_roles.yml", "roles");
+            success = success & uploadFile(esClient, cd+"sg_roles_mapping.yml", "rolesmapping");
+            success = success & uploadFile(esClient, cd+"sg_action_groups.yml", "actiongroups");
+            
+            ConfigUpdateResponse cur = esClient.execute(ConfigUpdateAction.INSTANCE, new ConfigUpdateRequest(new String[]{"config","roles","rolesmapping","actiongroups"})).actionGet();
+            
+            logger.debug("Succeed seeding SG? '{}'", success);
+            logger.debug("All node's config updated? '{}'", cur.getNodes().length == chr.getNumberOfNodes());
+        }
+	}
 	
+	private void seedInitialACL(Client esClient) throws Exception {
+
+		SearchGuardRoles roles = new SearchGuardRoles();
+		SearchGuardRolesMapping rolesMapping = new SearchGuardRolesMapping();
 		try {
-			//This should return nothing initially - if it does, we're done
-			acl = loadAcl(esClient);
+			if ( esClient.admin().indices().exists(new IndicesExistsRequest().indices(new String[]{"searchguard"})).actionGet().isExists() ) {
```

Comment:
I forgot about this, we should be pulling it from the configuration.  Currently SG doesn't let us specify the index it looks for [1]

[1] https://github.com/floragunncom/search-guard/issues/176